include/lsst/jointcal/Associations.h

```diff
@@ -135,11 +135,15 @@ class Associations {
      * @param[in]  matchCut       Separation radius to match fitted and
      *                            reference stars.
      * @param      fluxField      The field name in refCat to get the flux from.
+     * @param      refCoordinateErr Error on reference catalog coordinates (mas). If not NaN, this
+     *                              overrides the `coord_*_err` values in the reference catalog itself.
+     *                              This value is divided by cos(dec) before being used for ra_err.
      * @param      rejectBadFluxes  Reject reference sources with flux=NaN or 0 and/or fluxErr=NaN or 0.
      *                              Typically false for astrometry and true for photometry.
      */
     void collectRefStars(afw::table::SimpleCatalog &refCat, afw::geom::Angle matchCut,
-                         std::string const &fluxField, bool rejectBadFluxes = false);
+                         std::string const &fluxField, float const refCoordinateErr,
```

Comment:
Usually I prefer not to specify `const` for arguments passed by value but if this is your usual practice it may be fine too.
Another possible way to solve this NaN sentinel issue is to make an overload for `collectRefStars` with or without `refCoordinateErr` argument. With just one "optional" argument you need only two overloads which is not terrible,

Reply:
The idea of adding an overload is a good one, but I don't think it would cleanup the python calling code very much, sadly.